### PR TITLE
Implement Task Detailed View Redesign

### PR DESCRIPTION
**Pull request by jasperhuangg**
### Fixed Issues

$ https://github.com/Expensify/App/issues/19330
$ https://github.com/Expensify/App/pull/20841

### Tests/Offline tests/QA
These tests should function exactly the same when offline, since we're setting optimistic data for everything
1. FAB > Assign task. Finish the flow for creating the task. Or just navigate to an existing task.
2. Verify the UI looks as expected (see screenshots).
3. Click on the title item. Verify it opens a title input. Change the title to be a really long name. Verify the task title expands to contains the name, and then has ellipses if the name is too long.
4. Click on the description item. Verify it opens a description input. Change the description to be a really long name. Verify the task title expands to contains the name, and then has ellipses if the name is too long.
5. Assign the task to a person you have a chat with already. Verify that the assignee updates. (NOTE: there is currently a bug on main for assigning tasks to people you don't have chats with, so we won't test that here).
6. Click the checkmark to complete the task. Verify the text in the button above changes to "Mark as incomplete". Uncheck the checkmark. Verify the text in the button above changes to "Mark as done".
7. Repeat the same behavior with the button and verify the checkmark and button respond accordingly.

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

Screenshot 2023-06-22 at 6 08 52 PM

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Screenshot 2023-06-22 at 8 07 31 PM

</details>

<details>
<summary>Mobile Web - Safari</summary>

Screenshot 2023-06-22 at 8 07 10 PM

</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-06-22 at 6 11 12 PM

</details>

<details>
<summary>iOS</summary>

Screenshot 2023-06-22 at 8 02 33 PM

</details>

<details>
<summary>Android</summary>

Screenshot 2023-06-22 at 7 27 01 PM

</details>

### CODE REVIEW

File: src/components/LHNOptionsList/OptionRowLHN.js [hunk 1]
```diff
@@ -87,7 +87,7 @@ function OptionRowLHN(props) {
         !hasBrickError &&
         (optionItem.isUnreadWithMention ||
             (optionItem.hasOutstandingIOU && !optionItem.isIOUReportOwner) ||
-            (optionItem.isTaskReport && optionItem.isTaskAssignee && !optionItem.isTaskCompleted));
+            (optionItem.isTaskReport && optionItem.isTaskAssignee && !optionItem.isCompletedTaskReport));
```
Comment (thienlnam) [inline src/components/LHNOptionsList/OptionRowLHN.js:90]: Does this not work right now?

File: src/components/MenuItem.js [hunk 2]
```diff
@@ -97,6 +101,19 @@ function MenuItem(props) {
...
+            onPress={() => props.onPressSelection()}
+            isChecked={props.isSelected}
+            style={props.shouldShowSelectedStateBeforeTitle && styles.mr3}
+        />
+    ) : (
+        <SelectCircle
+            isChecked={props.isSelected}
+            style={props.shouldShowSelectedStateBeforeTitle && styles.mr3}
+        />
+    );
```
Comment (thienlnam) [inline src/components/MenuItem.js:115]: When does this show a circle?

Comment (thienlnam) [reply src/components/MenuItem.js:115]: Oh I see before this it would show the circle

File: src/libs/ReportUtils.js [hunk 3]
```diff
@@ -1067,7 +1091,7 @@ function getDMRootReportName(report) {
  * @returns {String}
  */
 function getChatRoomSubtitle(report) {
-    if (isThread(report)) {
+    if (isChildReport(report)) {
         if (!getChatType(report)) {
             return `${Localize.translateLocal('threads.from')} ${getDMRootReportName(report)}`;
         }
```
Comment (thienlnam) [inline src/libs/ReportUtils.js:1097]: No need to worry about the subtitle - will be taken care of in a different PR

File: src/components/LHNOptionsList/OptionRowLHN.js (hunk 1 above)
Comment (thienlnam) [reply src/components/LHNOptionsList/OptionRowLHN.js:90]: Ah I see you updated the method

File: src/components/TaskHeader.js [hunk 4]
```diff
@@ -1,123 +1,30 @@
-import React, {useEffect} from 'react';
```
Comment (thienlnam) [inline src/components/TaskHeader.js:1]: I think we would remove this file completely actually, the only thing this is doing is showing the call to action button right? This will also be needed for manual requests, so maybe we can merge this into Header and make it a little more generalized

**Review by thienlnam — CHANGES_REQUESTED** [review]

File: src/components/TaskHeader.js (hunk 4 above)
Comment (jasperhuangg) [reply src/components/TaskHeader.js:1]: done

**Comment by jasperhuangg** [comment]
There's a bug with editing the assignee of an existing task to someone you've never spoken to before @thienlnam, but that actually already happens on main so I don't think we should block this PR on it.

**Comment by thienlnam** [comment]
Yup agreed, makes sense to me

**Comment by shawnborton** [comment]
For this:
image

Can we vertically align the checkbox with the title and arrow? So that it looks like this:
image


For assignee:
image

...there should be 8px margin below the label Assignee and the avatar. The avatar should be 24px tall. 

We're also missing the empty state in the negative area above the task title:
image


For the horizontal rule style, we should only use that once a comment is added to the task. And then once it's added, it should not stretch full screen like you have it - it should have 20px padding on the left/right:
image

**Comment by jasperhuangg** [comment]
@shawnborton 

I've fixed the vertical alignment. It's still not going to be 100% vertically aligned for lowercase letters since we need to leave space in the line height for uppercase letters, which for this particular font are a lot larger than lowercase letters.
Screenshot 2023-06-23 at 10 57 25 AM
Screenshot 2023-06-23 at 10 58 34 AM

I also fixed the assignee like you asked
Screenshot 2023-06-23 at 11 32 58 AM

And the horizontal rule style
Screenshot 2023-06-23 at 11 43 24 AM

> We're also missing the empty state in the negative area above the task title:

Can you clarify what you mean by this?

**Comment by shawnborton** [comment]
Rather than vertically center the checkbox and the text and the arrow, we might want the arrow and checkbox to align to the top, so it feels more like a bulleted list item:
image


As for empty state, I mean the empty state illustration that you see when you are in a new chat. I think that's missing from threads in general, so we could always address that in a separate issue.

**Comment by jasperhuangg** [comment]
@shawnborton It's all vertical aligned now!

Screenshot 2023-06-23 at 11 52 43 AM

**Comment by jasperhuangg** [comment]
> As for empty state, I mean the empty state illustration that you see when you are in a new chat. I think that's missing from threads in general, so we could always address that in a separate issue.

I'll think it's better if we address this in a followup issue since it touches so many other things. I can create a follow up issue.

**Comment by shawnborton** [comment]
Nice, looks great! How big is the checkbox? Can you show screenshots of it checked as well?

**Comment by jasperhuangg** [comment]
@shawnborton here's it checked/unchecked

Screenshot 2023-06-23 at 12 12 07 PM
Screenshot 2023-06-23 at 12 12 02 PM

**Comment by shawnborton** [comment]
Nice. Can you confirm the size of the carets, too? They should be 20x20 but just want to double check.

**Comment by jasperhuangg** [comment]
@shawnborton Good looks, seems like they were 14px. I increased their size to 20px, but it didn't look right, and then checked the figma and I think they're supposed to be 16px.

#### 20px
Screenshot 2023-06-23 at 12 23 02 PM


#### 16px
Screenshot 2023-06-23 at 12 23 13 PM

**Comment by jasperhuangg** [comment]
@fedirjh @thienlnam this is ready for review!

**Review by fedirjh — COMMENTED** [review]
Left few comments

File: src/components/ReportActionItem/TaskView.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                title={props.report.description}
+                onPress={() => Navigation.navigate(ROUTES.getTaskReportDescriptionRoute(props.report.reportID))}
+                shouldShowRightIcon
+                disabled={!isOpen}
+                wrapperStyle={[styles.pv2]}
+            />
+            <MenuItem
+                label={props.translate('task.assignee')}
+                title={ReportUtils.getDisplayNameForParticipant(props.report.managerID)}
+                icon={UserUtils.getAvatar(
```
[the remark this answers is not in the document]
Comment (jasperhuangg) [reply src/components/ReportActionItem/TaskView.js:106]: Ah I think you may need a Web-E PR to handle this, I thought it was sorted on `main` so elected not to push the PR but seems it wasn't

**Comment by jasperhuangg** [comment]
> Also when the title is short the arrow does not align to the top.
> 
> Screenshot 2023-06-24 at 8 38 59 PM


@fedirjh I think everything is aligned how it's supposed to in this particular case:

Screenshot 2023-06-27 at 11 09 35 AM


Here's how the alignment looks in the Figma file. You can imagine that if the second line didn't exist then the alignment would be the same. The checkmark is intended to be vertically center-aligned with the first line of text.
Screenshot 2023-06-27 at 11 10 22 AM

File: src/components/ReportActionItem/TaskView.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+                onPress={Session.checkIfActionIsAllowed((e) => {
+                    if (e && e.type === 'click') {
+                        e.currentTarget.blur();
+                    }
+
+                    Navigation.navigate(ROUTES.getTaskReportTitleRoute(props.report.reportID));
+                })}
+                style={({hovered, pressed}) => [styles.ph5, styles.pv2, StyleUtils.getButtonBackgroundColorStyle(getButtonState(hovered, pressed), true)]}
+                ref={props.forwardedRef}
+                disabled={!isOpen}
```
[the remark this answers is not in the document]
Comment (jasperhuangg) [reply src/components/ReportActionItem/TaskView.js:62]: fixed!

File: src/components/ReportActionItem/TaskView.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (jasperhuangg) [reply src/components/ReportActionItem/TaskView.js:107]: Holding on https://github.com/Expensify/Web-Expensify/pull/37967, which should fix this

**Comment by jasperhuangg** [comment]
> We have another crash
> 
> Screenshot 2023-06-24 at 8 32 38 PM

fixed this!

**Comment by jasperhuangg** [comment]
@fedirjh good for another look! Thanks again in advance for your review :)

**Comment by jasperhuangg** [comment]
> > I think everything is aligned how it's supposed to in this particular case:
> 
> @jasperhuangg check the video , hover over the task title, it's noticeable that that the arrow is not centered, on the other hand the assignee arrow is centered
> 
>  Screen.Recording.2023-06-27.at.7.39.21.PM.mov

@shawnborton what do you think about this? I thought the arrow was supposed to be vertically center aligned with the text/checkbox. And since they aren't vertically center aligned with the entire component, then the arrow won't be either.

**Review by fedirjh — COMMENTED** [review]
LHN doesn't display latest actions , mark as done, reopen task, task assigned to someone else 

https://github.com/Expensify/App/assets/36869046/16d7b61a-fc2b-44c8-9cfd-08cd5875a8d8

File: src/pages/tasks/TaskDescriptionPage.js
[the remark this answers is not in the document]
Comment (jasperhuangg) [reply src/pages/tasks/TaskDescriptionPage.js:1]: I'm unsure, but we don't need to address that now. That can be part of a cleanup issue later on.

**Review by fedirjh — COMMENTED** [review]
Few comment

File: src/pages/home/ReportScreen.js [hunk 7]
```diff
@@ -293,11 +293,14 @@ class ReportScreen extends React.Component {
...
-                                    personalDetails={this.props.personalDetails}
-                                />
+                            {ReportUtils.isTaskReport(this.props.report) && this.props.isSmallScreenWidth && (
+                                <View style={styles.borderBottom}>
+                                    <View style={[styles.darkBG, styles.pl0]}>
+                                        <View style={[styles.ph5, styles.pb3]}>
+                                            <TaskHeaderActionButton report={this.props.report} />
+                                        </View>
+                                    </View>
+                                </View>
```
[the remark this answers is not in the document]
Comment (jasperhuangg) [reply src/pages/home/ReportScreen.js:303]: yes, good catch. this was copied over from a different file, I'll change it!

File: src/pages/home/ReportScreen.js [hunk 8]
```diff
@@ -293,11 +293,14 @@ class ReportScreen extends React.Component {
...
                             )}
 
-                            {ReportUtils.isTaskReport(this.props.report) && (
-                                <TaskHeader
-                                    report={this.props.report}
-                                    personalDetails={this.props.personalDetails}
-                                />
+                            {ReportUtils.isTaskReport(this.props.report) && this.props.isSmallScreenWidth && (
+                                <View style={styles.borderBottom}>
+                                    <View style={[styles.darkBG, styles.pl0]}>
```
[the remark this answers is not in the document]
Comment (jasperhuangg) [reply src/pages/home/ReportScreen.js:298]: hmmm interesting, this never happened for me

[the remark this answers is not in the document]
Comment (jasperhuangg) [reply src/pages/home/ReportScreen.js:298]: you're on an iOS sim? it appears to render fine for me
Screenshot 2023-07-05 at 2 27 23 PM

[the remark this answers is not in the document]
Comment (jasperhuangg) [reply src/pages/home/ReportScreen.js:298]: ahh interesting, that only happens when you combine the views that you're talking about [here](https://github.com/Expensify/App/pull/20841/files#r1253572989)

File: src/pages/home/ReportScreen.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (jasperhuangg) [reply src/pages/home/ReportScreen.js]: See https://github.com/Expensify/App/pull/20841#discussion_r1253674721 for more

File: src/pages/home/ReportScreen.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (jasperhuangg) [reply src/pages/home/ReportScreen.js]: fixed

**Comment by thienlnam** [comment]
Appears to have a few conflicts with the latest header updates

**Comment by thienlnam** [comment]
cc @fedirjh When you get a chance could you review today? Would love to get this out - thank you!

**Comment by jasperhuangg** [comment]
@thienlnam we've been chatting in the Slack DM working through the review! He's completing the reviewer checklist as we speak.

**Comment by thienlnam** [comment]
Oh lovely - thank you guys!

**Review by fedirjh — APPROVED** [review]
LGTM and tests well 🚀

File: src/components/MenuItem.js [hunk 9]
```diff
@@ -63,10 +63,13 @@ const defaultProps = {
     floatRightAvatars: [],
     shouldStackHorizontally: false,
     avatarSize: undefined,
+    floatRightAvatarSize: undefined,
     shouldBlockSelection: false,
     hoverAndPressStyle: [],
     furtherDetails: '',
     furtherDetailsIcon: undefined,
+    isSmallAvatarSubscriptMenu: false,
+    title: '',
```
Comment (thienlnam) [inline src/components/MenuItem.js:72]: I don't see title used here

**Review by thienlnam — COMMENTED** [review]
Code looking solid! Just going to pull and test a bit

**Comment by thienlnam** [comment]
Screenshot 2023-07-06 at 4 31 36 PM

Noticed this in testing - when it's completed you can still highlight over the title which looks odd

**Comment by thienlnam** [comment]
But other than that it tests great and looks awesome - nice work!

**Comment by thienlnam** [comment]
Also while you're here - could you please update the Cancelled Task parameter optimisticReportActionID to cancelledTaskReportActionID

**Comment by jasperhuangg** [comment]
> Screenshot 2023-07-06 at 4 31 36 PM
> 
> 
> 
> Noticed this in testing - when it's completed you can still highlight over the title which looks odd
> 
> 

Ah hmmm we fixed this, maybe it wasn't pushed, lemme check

File: src/libs/actions/Task.js [hunk 10]
```diff
@@ -624,11 +625,7 @@ function cancelTask(taskReportID, taskTitle, originalStateNum, originalStatusNum
         },
     ];
 
-    API.write('CancelTask', {taskReportID, optimisticReportActionID}, {optimisticData, failureData});
-}
-
-function isTaskCanceled(taskReport) {
-    return taskReport.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && taskReport.statusNum === CONST.REPORT.STATUS.CLOSED;
+    API.write('CancelTask', {, cancelledTaskReportActionID: optimisticReportActionID, taskReportID}, {optimisticData, failureData});
```
Comment (thienlnam) [inline src/libs/actions/Task.js:628]: Shouldn't have the comma here

**Review by thienlnam — CHANGES_REQUESTED** [review]

**Comment by shawnborton** [comment]
@thienlnam how did you get this screenshot where the description and assignee are gray?
image


Also, let me know when we have final final screenshots for testing. Perhaps we can create a test build too?

**Comment by shawnborton** [comment]
I'm a little confused by this:
image


If you click the checkbox to complete a task, it should be the same as clicking the green button up top. 

And then, why do things gray out when the task is complete?

File: src/libs/actions/Task.js (hunk 10 above)
Comment (jasperhuangg) [reply src/libs/actions/Task.js:628]: I'm so confused, that comma isn't there for me locally 😕

File: src/components/MenuItem.js (hunk 9 above)
Comment (jasperhuangg) [reply src/components/MenuItem.js:72]: it's used below

**Comment by jasperhuangg** [comment]
@shawnborton Here's how it looks when it's canceled:

Screenshot 2023-07-07 at 11 16 45 AM

And here's how it looks when it's disabled:
Screenshot 2023-07-07 at 11 16 54 AM

**Comment by shawnborton** [comment]
Cool, much better. So then why do we have a green button in the header when the checkbox is already checked?

**Comment by jasperhuangg** [comment]
> Cool, much better. So then why do we have a green button in the header when the checkbox is already checked?

Ah hmm didn't realize that wasn't supposed to be there. The text changes to "Mark as incomplete" when the checkbox is checked. But I can remove it.

**Comment by jasperhuangg** [comment]
Screenshot 2023-07-07 at 11 26 16 AM

**Comment by shawnborton** [comment]
Yeah, my understanding is that hitting the checkbox or hitting the green button does the same thing.

**Comment by jasperhuangg** [comment]
> Yeah, my understanding is that hitting the checkbox or hitting the green button does the same thing.

What do you mean by that? In how it was before, they would do the same thing. Checking the checkbox changes the button text from "mark as done" to "mark as incomplete". Unchecking the checkbox changes it back. And clicking the button does vice versa.

**Comment by shawnborton** [comment]
Got it, then I think we're on the same page. But the idea is that the checkbox basically does the same thing as the header buttons do.

**Review by thienlnam — DISMISSED** [review]
Awesome, thank you!

**Review by thienlnam — APPROVED** [review]
Merging this quickly with haste since it's a big PR that continues to get merge conflicts

Design comments have been addressed

**Comment by ayazhussain79** [comment]
App crashes when creating task  https://expensify.slack.com/archives/C049HHMV9SM/p1688766194182809

File: src/components/ReportActionItem/TaskView.js [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+                shouldGreyOutWhenDisabled={false}
+            />
+            {props.report.managerID ? (
+                <MenuItem
+                    label={props.translate('task.assignee')}
+                    title={ReportUtils.getDisplayNameForParticipant(props.report.managerID)}
+                    icon={UserUtils.getAvatar(
+                        PersonalDetailsUtils.getPersonalDetailsByIDs([props.report.managerID], props.currentUserPersonalDetails.accountID)[0].avatar,
+                        props.report.managerID,
+                    )}
```
Comment (s77rt) [inline src/components/ReportActionItem/TaskView.js:118]: This caused a regression https://github.com/Expensify/App/issues/22657. `PersonalDetailsUtils.getPersonalDetailsByIDs` could return an empty array e.g. for users that you have not interacted with before (no personaldetails). Then we will be trying to access property `avatar` of `undefined`.

File: src/pages/home/report/ReportActionItem.js [hunk 12]
```diff
@@ -541,6 +554,17 @@ export default compose(
...
+            prevProps.action.actionName === CONST.REPORT.ACTIONS.TYPE.CREATED &&
+            ReportUtils.isTaskReport(nextProps.report) &&
+            nextProps.action.actionName === CONST.REPORT.ACTIONS.TYPE.CREATED &&
+            prevProps.report.reportName === nextProps.report.reportName &&
+            prevProps.report.description === nextProps.report.description &&
+            ReportUtils.isCompletedTaskReport(prevProps.report) === ReportUtils.isCompletedTaskReport(nextProps.report) &&
+            prevProps.report.managerID === nextProps.report.managerID &&
+            prevProps.report.managerEmail === nextProps.report.managerEmail &&
+            prevProps.isOnlyReportAction === nextProps.isOnlyReportAction,
     ),
```
Comment (StevenKKC) [inline src/pages/home/report/ReportActionItem.js:569]: Hi ✋ Coming from https://github.com/Expensify/App/issues/24267

We should add comparing of `report.amount` to dynamically update total amount in money request view.

Comment (jasperhuangg) [reply src/pages/home/report/ReportActionItem.js:569]: Hmm that doesn't relate to tasks, which this PR implements, but good thinking! That seems like that would solve the problem.

File: src/components/ReportActionItem/TaskView.js [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+    }, [props.report]);
+
+    const taskTitle = convertToLTR(props.report.reportName || '');
+    const isCompleted = ReportUtils.isCompletedTaskReport(props.report);
+    const isOpen = ReportUtils.isOpenTaskReport(props.report);
+    const isCanceled = ReportUtils.isCanceledTaskReport(props.report);
+
+    return (
+        <View>
+            <PressableWithSecondaryInteraction
```
Comment (situchan) [inline src/components/ReportActionItem/TaskView.js:53]: What was reason for using `PressableWithSecondaryInteraction` in `TaskView` instead of just `PressableWithFeedback`?
I don't see any secondary interaction.
